dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/ORCIDHelper.java

```diff
@@ -0,0 +1,127 @@
+            Url url = new Url(v.getDoiURL());
+            externalID.setUrl(url);
+            externalID.setValue(v.getDoiURL());
+            title.setContent(e.getEntryPath() + ":" + v.getName());
+        } else {
+            Url url = new Url(e.getConceptDoi());
+            externalID.setUrl(url);
+            externalID.setValue(e.getConceptDoi());
+            title.setContent(e.getEntryPath());
+        }
+        Title journalTitle = new Title();
+        journalTitle.setContent("Dockstore");
+        work.setJournalTitle(journalTitle);
+        workTitle.setTitle(title);
+        workTitle.setSubtitle(subtitle);
+        work.setWorkTitle(workTitle);
+        work.setShortDescription("A workflow exported from Dockstore");
+        work.setWorkType(WorkType.SOFTWARE);
+        externalID.setRelationship(Relationship.SELF);
+        externalIDs.getExternalIdentifier().add(externalID);
+        work.setWorkExternalIdentifiers(externalIDs);
+        GregorianCalendar gregory = new GregorianCalendar();
+        gregory.setTime(new Date());
+        XMLGregorianCalendar calendar = DatatypeFactory.newInstance()
+                .newXMLGregorianCalendar(
+                        gregory);
+        CreatedDate createdDate = new CreatedDate(calendar);
+        work.setCreatedDate(createdDate);
+        LastModifiedDate lastModifiedDate = new LastModifiedDate();
+        lastModifiedDate.setValue(calendar);
+        work.setLastModifiedDate(lastModifiedDate);
+        return transformWork(work);
+    }
+
+    public static HttpResponse postSandboxWorkString(String id, String workString, String token) throws IOException {
+        return postWorkString(ORCID_SANDBOX_BASE_URL, id, workString, token);
+    }
+
+    public static HttpResponse postProdWorkString(String id, String workString, String token) throws IOException {
+        return postWorkString(ORCID_BASE_URL, id, workString, token);
+    }
+
+
+    private static String transformWork(Work work) throws JAXBException {
+        JAXBContext context = JAXBContext.newInstance(Work.class);
+        StringWriter writer = new StringWriter();
+        Marshaller marshaller = context.createMarshaller();
+        marshaller.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, Boolean.TRUE);
+        marshaller.marshal(work, writer);
+        return writer.getBuffer().toString();
+    }
+
+    private static HttpResponse postWorkString(String baseURL, String id, String workString, String token) throws IOException {
+        HttpPost postRequest = new HttpPost(baseURL + id + "/work");
+        postRequest.addHeader("content-type", "application/vnd.orcid+xml");
+        postRequest.addHeader("Authorization", "Bearer " + token);
+        StringEntity workEntity = new StringEntity(workString);
+        postRequest.setEntity(workEntity);
+        CloseableHttpClient build = HttpClientBuilder.create().build();
+        HttpResponse response = build.execute(postRequest);
+        return response;
```

Comment:
I think this needs to be [closed](https://stackoverflow.com/questions/21574478/what-is-the-difference-between-closeablehttpclient-and-httpclient-in-apache-http). But if you close it, can you still access the response?

I know this is in progress, but ultimately it needs to be solved by either returning something else or making clear in the JavaDoc that the caller of this method needs to close the response.